src/types/helperTypes.ts

```diff
@@ -122,3 +122,15 @@ export type DataSourceForPlaceable<P extends PlaceableObject> = P extends Placea
     ? D['_source']
     : never
   : never;
+
+/**
+ * If `T` is `Promise<TResult>` then `TResult`; otherwise `T`.
+ */
+export type PromisedType<T> = T extends Promise<infer TResult> ? TResult : T;
+
+/**
+ * If `ReturnType<T>` is `Promise<TResult>` then `TResult`; otherwise `ReturnType<T>`.
+ */
+export type PromisedReturnType<T extends (...args: any) => any> = ReturnType<T> extends Promise<infer TResult>
```

Comment:
Isn't this the same as
```ts
export type PromisedReturnType<T extends (...args: any) => any> = PromisedType<ReturnType<T>>;
```
?

Reply:
You're right, I don't know what I was thinking.